src/overlays/actors/ovl_Bg_Breakwall/z_bg_breakwall.c

```diff
@@ -53,21 +66,268 @@ static ColliderQuadInit D_80870760 = {
+    }
+}
+
+void BgBreakwall_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    BgBreakwall* this = THIS;
+
+    DynaPoly_DeleteBgActor(globalCtx, &globalCtx->colCtx.dyna, this->dyna.bgId);
+}
+
+/**
+ * Spawns fragments using ACTOR_EN_A_OBJ whenever the wall or floor is exploded.
+ * Although it is not ever checked, it will also return the instance of the actor which it spawns.
+ */
+Actor* BgBreakwall_SpawnFragments(GlobalContext* globalCtx, BgBreakwall* this, Vec3f* pos, f32 velocity, f32 scaleY,
+                                  f32 scaleX, s32 count, f32 accel) {
+    Actor* actor;
+    Vec3f actorPos;
+    s32 k;
+    s32 j;
+    s32 i;
+    s16 angle1;
+    s16 angle2 = 0;
+    Vec3f zeroVec = { 0.0f, 0.0f, 0.0f }; // unused
+    Vec3s actorRotList[] = { { 0, 0, 0 }, { 0, 0, 0x4000 }, { 0, 0, -0x4000 }, { 0, 0, 0 } };
+    Vec3f actorScaleList[] = {
+        { 0.004f, 0.004f, 0.004f },
+        { 0.004f, 0.004f, 0.004f },
+        { 0.004f, 0.004f, 0.004f },
+        { 0.004f, 0.004f, 0.004f },
+    };
+    Vec3f actorPosList[][4] = {
+        { { 40.0f, 15.0f, 0.0f }, { 30.0f, 57.0f, 0.0f }, { 50.0f, 57.0f, 0.0f }, { 40.0f, 70.0f, 0.0f } },
+        { { 55.0f, -15.0f, 0.0f }, { 30.0f, -32.0f, 0.0f }, { 50.0f, -32.0f, 0.0f }, { 20.0f, -10.0f, 0.0f } },
+        { { -40.0f, 14.0f, 0.0f }, { -50.0f, 57.0f, 0.0f }, { -30.0f, 57.0f, 0.0f }, { -40.0f, 70.0f, 0.0f } },
+        { { -55.0f, -15.0f, 0.0f }, { -55.0f, -32.0f, 0.0f }, { -30.0f, -32.0f, 0.0f }, { -20.0f, -10.0f, 0.0f } },
+    };
+    s32 pad;
+
+    for (k = 3; k >= 0; k--) {
+        if ((k == 0) || (k == 3)) {
+            actorScaleList[k].x *= scaleX;
+            actorScaleList[k].y *= scaleY;
+            actorScaleList[k].z *= scaleY;
+        } else {
+            actorScaleList[k].x *= scaleY;
+            actorScaleList[k].y *= scaleX;
+            actorScaleList[k].z *= scaleX;
+        }
+    }
+
+    for (i = 0; i < count; angle2 += 0x4000, i++) {
+        angle1 = ABS(this->dyna.actor.world.rot.y) + angle2;
+        Matrix_Translate(this->dyna.actor.world.pos.x, this->dyna.actor.world.pos.y, this->dyna.actor.world.pos.z,
+                         MTXMODE_NEW);
+        Matrix_RotateRPY(this->dyna.actor.world.rot.x, this->dyna.actor.world.rot.y, this->dyna.actor.world.rot.z,
+                         MTXMODE_APPLY);
+        Matrix_Translate(pos->x, pos->y, pos->z, MTXMODE_APPLY);
+
+        for (j = 3; j >= 0; j--) {
+            for (k = 3; k >= 0; k--) {
+                Matrix_MultVec3f(&actorPosList[j][k], &actorPos);
+                actor =
+                    Actor_Spawn(&globalCtx->actorCtx, globalCtx, ACTOR_EN_A_OBJ, Rand_CenteredFloat(20.0f) + actorPos.x,
+                                Rand_CenteredFloat(20.0f) + actorPos.y, Rand_CenteredFloat(20.0f) + actorPos.z,
+                                actorRotList[k].x, actorRotList[k].y + angle1, actorRotList[k].z, 0x000B);
+
+                if ((j & 1) == 0) {
```

Comment:
```suggestion
                if ((j % 2) == 0) {
```
does this work?

Reply:
It didn't earlier but I'll try it again since now we have the function fully matched.

Reply:
nope